ambari-server/src/main/java/org/apache/ambari/server/state/UpgradeContext.java

```diff
@@ -387,7 +387,8 @@ public UpgradeContext(@Assisted Cluster cluster,
         throw new AmbariException(message);
       }
 
-      m_repositoryVersion = priors.iterator().next();
+      // the "associated" repository of the revert is the target of what's being reverted
+      m_repositoryVersion = revertUpgrade.getRepositoryVersion();
```

Comment:
The associated repository for a revert should be the same associated repository as the upgrade being reverted:

Upgrade: From ZK 2.5.1.0, Storm 2.5.1.1; To 2.5.2.0 (associated is 2.5.2.0)
Revert: From 2.5.2.0 to ZK 2.5.1.0, Storm 2.5.1.1 (associated is still 2.5.2.0 since it's the "from" repo)

This change in revert caused a few other incidental changes sprinkled throughout the codebase.